examples/product-detail-page/src/components/ProductView.vue

```diff
@@ -0,0 +1,260 @@
+<script setup lang="ts">
+  import { computed, unref, defineProps, ref } from "vue";
+  import {
+    getProductName,
+    getSmallestThumbnailUrl,
+    getSrcSetForMedia,
+  } from "@shopware-pwa/helpers-next";
+  import {
+    useAddToCart,
+    useProductPrice,
+    usePrice,
+    useProductAssociations,
+  } from "@shopware-pwa/composables-next";
+  import ProductCard from "./ProductCard.vue";
+
+  const props = defineProps(["product"]);
+  const product = computed(() => props.product);
+
+  const activeTab = ref<
+    "product-description" | "product-properties" | "product-cross-sell"
+  >("product-description");
+
+  const { addToCart } = useAddToCart(product);
+  const { loadAssociations, isLoading, productAssociations } =
+    useProductAssociations(product, {
+      associationContext: "cross-selling",
+    });
+  const openRecommendedTab = () => {
+    activeTab.value = "product-cross-sell";
+    loadAssociations({
```

Comment:
we're invoking it also when you switch between tabs, couldn't we load it only once when tab is opened for the first time?

Reply:
yes, it's being invoked every time. I can improve it for sharing good practices of course :)